internal/handlers/registry/registry_test.go

```diff
@@ -14,8 +14,62 @@
 
 package registry
 
-import "testing"
+import (
+	"encoding/json"
+	"os/exec"
+	"testing"
 
-func TestDummy(t *testing.T) {
-	// we need at least one test per package to correctly calculate coverage
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/maps"
+)
+
+// extraDeps returns extra FerretDB dependencies (packages) when the given tag is enabled.
+func extraDeps(t *testing.T, tag string) []string {
+	t.Helper()
+
+	type goListResult struct {
+		Deps []string `json:"Deps"`
+	}
+
+	b, err := exec.Command("go", "list", "-json", "../../../cmd/ferretdb").Output()
+	require.NoError(t, err)
+	var withoutTag goListResult
+	require.NoError(t, json.Unmarshal(b, &withoutTag))
+
+	b, err = exec.Command("go", "list", "-json", "-tags", tag, "../../../cmd/ferretdb").Output()
+	require.NoError(t, err)
+	var withTag goListResult
+	require.NoError(t, json.Unmarshal(b, &withTag))
+
+	diff := make(map[string]struct{}, len(withTag.Deps))
+	for _, dep := range withTag.Deps {
+		diff[dep] = struct{}{}
+	}
+
+	for _, dep := range withoutTag.Deps {
+		delete(diff, dep)
+	}
+
+	return maps.Keys(diff)
+}
+
+// TestDeps ensures that extra dependencies are not used when the given Go build tag / backend is not enabled.
```

Comment:
The test seems to test extra dependencies are enabled instead of _not enabled_. Perhaps we want to update this doc string :hugs:

Reply:
Well, the motivation for that test is to ensure that dependencies of disabled handlers are not imported somewhere else by accident

Reply:
PTAL

Reply:
Good intention thought :hugs: